File: Sources/AINotebookCore/LocaleDetection.swift
/// Maps the user's preferred-language list (from `Locale.preferredLanguages`)
/// to an `AppLanguage`. Czech if any preferred entry starts with "cs",
/// otherwise English.
///
/// Pulled out into a free function so it can be tested without touching the
/// real `Locale` singleton.
public func detectInitialLanguage(preferredLanguages: [String]) -> AppLanguage {
    for entry in preferredLanguages {
        if entry.lowercased().hasPrefix("cs") {
            return .czech
        }
    }
    return .english
}

File: windows/src/AINotebook.Core/LocaleDetection.cs
using AINotebook.Core.Models;

namespace AINotebook.Core;

public static class LocaleDetection
{
    /// Czech if any preferred entry starts with "cs" (case-insensitive),
    /// otherwise English.
    public static AppLanguage DetectInitialLanguage(IEnumerable<string> preferred)
    {
        foreach (var entry in preferred)
        {
            if (entry.StartsWith("cs", StringComparison.OrdinalIgnoreCase))
                return AppLanguage.Czech;
        }
        return AppLanguage.English;
    }
}
